Fix #3332: Rescale Lesson Avatar Image

shaz13 opened the pull request:
#3332
Rescaled the Avatar as stated in the issue. 

**PR's commit** 

![shot 06](https://cloud.githubusercontent.com/assets/24438869/25013482/5985d88a-2091-11e7-99e2-3d600c03e2cc.jpg)

shaz13 commented:
Few Notes: 
1. Please review the PR and suggest the size (height and width)
2. Current commit address the Mockup as presented in the issue. But, locally user-avatar has been modified too. Once, size is finalised, will push the next commit. (Just so to keep the commits clean)

Local Changes (Mock Up)
![shot 07](https://cloud.githubusercontent.com/assets/24438869/25013752/7ad125ca-2092-11e7-93d6-988595953f0e.jpg)

shaz13 commented:
Latest commit changes: 
![shot 09](https://cloud.githubusercontent.com/assets/24438869/25014687/63520c94-2096-11e7-91d9-70c5c43af93c.jpg)

markhalpin commented:
hi @shaz13, looking good. A couple of things:
Let's try going a little bit bigger. Also, could you use the actual Otter icon to test, since that's a complex image and we should see if it's big enough to read?
For the user avatar outline, use our darkest green (the one from the navbar at bottom of the page). That's dark enough to be neutral and so should look decent with any user image.
Can you drop the upper left avatar down a bit relative to the top of the page. Having a little air there, as opposed to flushing it to the top, looks a bit more "open". Refer to original mock.
I'm heading into meetings so might not respond again until later this evening. Thanks.

shaz13 commented:
Added the cute Otter to the mock. And, corrected top margins. And regarding making it bigger, in my local tests, image is getting blury beyond 80px as the avatar is of 72px. 
![shot 10](https://cloud.githubusercontent.com/assets/24438869/25035840/b86859f0-210d-11e7-9382-df5a729584f1.jpg)

markhalpin commented:
ahhhh, that's part of the legibility problem. Maybe we should be using an .svg for this, as with the other vector graphics sitewide (like banners etc.)? That would keep it crisp at any size. There should be some floating around, but let me know if you'd just like me to send you one. An ideal case would be something like the page banners, where we have three different .svgs (the avatar in different poses or expressions) and they're randomly deployed. Let me know if I should send you these. Thanks.

shaz13 commented:
@markhalpin The local development server has **png** images but the production uses **svg** format. I have managed to extract it from the production server and the mock uses that extracted Otter **svg** but didn't push commit as it was more of a hack. And, the size 80px should do well in my opinion. Making it larger feels like card and text are low scaled compared to avatar. What do you think?  Awaiting your further instructions.Thanks!

markhalpin commented:
Go ahead and commit it at that size and we'll see how it works. I would suggest swapping in the .svg though (if I understand what you're saying). Thanks.

shaz13 commented:
@markhalpin  Replaced the image with Otter and made it bigger :smile_cat: Kindly check the commit

shaz13 commented:
Current commit changes: 
![shot 11](https://cloud.githubusercontent.com/assets/24438869/25062503/cee9defa-21ea-11e7-924f-275ab5956caa.jpg)

markhalpin commented:
Visually LGTM, thanks! You'll want to get someone else to sign off on the code.

shaz13 commented:
@seanlip  Kindly review the PR

shaz13 commented:
Yes, the Otter image is seen on production server only. The previous review
comments suggested to adopt otter image to give clear idea about size of
chat head. Hence otter image was placed in the commits.

On 19-Apr-2017 7:44 AM, "Xinyu Wu" <notifications@github.com> wrote:

> *@wxyxinyu* requested changes on this pull request.
>
> Taking over the review. Is the otter image supposed to be in develop? I
> think we use it only for production?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/3337#pullrequestreview-33393052>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AXToVYqYMZpldE63i3fvaafEcDwmJEUgks5rxW4XgaJpZM4M85bx>
> .
>

wxyxinyu commented:
Are you actually changing the image itself? We currently don't put the otter image in our codebase on github, instead it's added when a new version is deployed.

shaz13 commented:
Yes, current PR replaced the image. Shall it be reverted?


On 20-Apr-2017 9:27 AM, "Xinyu Wu" <notifications@github.com> wrote:

> Are you actually changing the image itself? We currently don't put the
> otter image in our codebase on github, instead it's added when a new
> version is deployed.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/3337#issuecomment-295570972>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AXToVbqTilGMm3ZL2s-Fdysp2cjBbs5Bks5rxteQgaJpZM4M85bx>
> .
>

wxyxinyu commented:
I mean, did you e.g. use any image editing software to edit the image, or is it the same image as the one on production?

shaz13 commented:
No. I  just deleted the old image and replaced the otter image with the
same file name.




On 20 April 2017 at 21:04, Xinyu Wu <notifications@github.com> wrote:

> I mean, did you e.g. use any image editing software to edit the image, or
> is it the same image as the one on production?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/3337#issuecomment-295782541>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AXToVSDKWjvAdfjg8DONXbXFcy87K8x7ks5rx3sEgaJpZM4M85bx>
> .
>

markhalpin commented:
Stepping back for a moment to try to clarify matters and avoid confusion that might be arising here. Earlier in the process, Shaz had made the comment "And regarding making it bigger, in my local tests, image is getting blury beyond 80px as the avatar is of 72px."This made me think he was suggesting this as a reason for not implementing the avatar at a larger size. I wrote back that if we were using an .svg image as opposed to a raster image, that shouldn't be an issue. It looks like we're using a low-res .png on the site, and we should replace that with an .svg, because yes, it's looking blurry even at a small size.

markhalpin commented:
I had also mentioned a few messages ago that I think we should do something similar to what we do with the icon banner; namely, having the avatar be deployed randomly from a choice of four or five poses, so that the image feels a bit more animated. Is that doable? If so, please let me know as I can create the various .svgs from my original files.

wxyxinyu commented:
@markhalpin -- yep, that's doable. So it seems like the avatar used on the site is not the same as the image @shaz13 has committed -- that image is used somewhere else. The one being used has a white background, and is indeed a png at 72px. @markhalpin if you could send an svg for that that'll be great.

shaz13 commented:
@wxyxinyu  Yes, the avatar on the production server is different. The current commit only reflects the change of avatar at **develop** branch. If PR is being merged and later released, the actual SVG would show up not this commit image. 
The image we modify in this PR's scope only reflects the develop branch not at production server even after merging.
And a question, @markhalpin the **svg** and animation comment, guess that is handled somewhere between **develop** and while releasing the version. 
@wxyxinyu  Am I right?

markhalpin commented:
Hi @shaz13 and @wxyxinyu I've placed four .svgs here: https://drive.google.com/open?id=0B4A9S7Ym8W0CTlZzNlVlWDJrMkU

Those should be used randomly each time the avatar is deployed. @shaz13 I don't literally mean animation, just more like in the sense of variety. I can file a separate issue for the randomization if that makes sense, but it seems like it might be just as wise to account for it in whatever you're doing here. Let me know.

Thanks!

markhalpin commented:
Oops, sorry, this link please.

https://drive.google.com/open?id=0B4A9S7Ym8W0CTlZzNlVlWDJrMkU

seanlip commented:
Avatar randomization should happen in a separate issue. Four versions of the black-O-in-the-circle should be used (it's fine if they're all the same); these will be substituted at deploy time.

This PR (and others) should stick to the black O with the circle, but please make an svg version of this and rename the file to exploration_otter.svg instead. We'll substitute the otter in at deploy time. This is done so that fresh deployments of Oppia by others have a clearly different identity.

shaz13 commented:
@markhalpin Thanks for the images. 
@wxyxinyu A small clarification before the next push. 
Could you explain how develop branch versions are pushed to a release? In develop branch the otter image is absent but we could see it on the production server. So, if I make any further commits will it be valid or things do change when it's being released on the production server. 

Example: 
- I am going to include otter1.png, otter2.png and otter3.png in upcoming commits and changing the respective image `ahref` scope to randomise the images.
- As, some files are replaced before release, Will it affect randomization? Kindly explain the **develop**, **changes in between** and **release** process?

shaz13 commented:
@seanlip Shall the present commits' otter be removed and replaced with the former oppia_black_72px.png image but in SVG format? Then the PR is good to go. Kindly confirm

seanlip commented:
> @seanlip Shall the present commits' otter be removed and replaced with the former oppia_black_72px.png image but in SVG format? Then the PR is good to go. Kindly confirm

Yes.

For randomization (later), you can create exploration_otter1.svg etc. in the same folder and randomly select between them, but use the black-o-in-white-circle image instead (I initially thought they could all be the same, but you might want to do slight variants so that we can make sure the randomization actually works). You'll just need to convert the existing png to svg.

shaz13 commented:
@seanlip Replaced the current PNG with SVG, also made necessary scope url corrections. 
Kindly check the commit.

markhalpin commented:
Thanks @seanlip. @wxyxinyu or @shaz13, would one of you please file the additional issue? I'm afraid I'm on a loaner laptop following a meltdown of my own, and am nowhere near having reset all my various webpages etc. Thanks!

shaz13 commented:
@markhalpin Sure, I will do it.

seanlip commented on core/templates/dev/head/pages/exploration_player/conversation_skin_directive.html:
```diff
@@ -157,12 +157,15 @@
       height: 36px;
       position: absolute;
-      top: 0;
+      top: 0px;
```
Here and elsewhere, just "0" is fine (no need units if the value is 0).

shaz13 replied:
Done.

seanlip commented on core/templates/dev/head/pages/exploration_player/conversation_skin_directive.html:
```diff
@@ -157,12 +157,15 @@
       width: 36px;
       z-index: 1;
     }
     .conversation-skin-oppia-avatar {
```
This appears to be defined twice -- once here, and once in tutor_card_directive.html. Can it be defined just once to avoid duplication -- or, if this is referring to something specific, could the name be changed to reflect just that specific element rather than all conversation-skin-oppia-avatars?

shaz13 replied:
Made it more specific. Thanks!

seanlip commented on assets/images/avatar/oppia_black_100px.svg:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Let's rename the file to oppia_avatar_100px.svg or similar, since the replaced avatar may not be black.

shaz13 replied:
Renamed to `oppia_avatar_100px.svg`

seanlip commented on core/templates/dev/head/pages/exploration_player/conversation_skin_directive.html:
```diff
@@ -161,8 +161,11 @@
     }
-    .conversation-skin-oppia-avatar {
-      left: -20px;
+    .conversation-skin-supplemental-card-container, .conversation-skin-oppia-avatar {
```
Just to check, is the comma intentional here?

shaz13 replied:
Yes, saw the comma practise at all CSS properties in the file. Hence added it too.

shaz13 replied:
Done, please have a look at latest commit

seanlip commented:
Comma and no comma mean completely different things in CSS. One implies
multiple selectors and the other implies descendant. So it is very
important to make sure that you are writing what you mean, and I just
wanted to double check that that is the case.

On Friday, April 21, 2017, Shahebaz <notifications@github.com> wrote:

> *@shaz13* commented on this pull request.
> ------------------------------
>
> In core/templates/dev/head/pages/exploration_player/
> conversation_skin_directive.html
> <https://github.com/oppia/oppia/pull/3337#discussion_r112802361>:
>
> > @@ -161,8 +161,11 @@
>        width: 36px;
>        z-index: 1;
>      }
> -    .conversation-skin-oppia-avatar {
> -      left: -20px;
> +    .conversation-skin-supplemental-card-container, .conversation-skin-oppia-avatar {
>
> Yes, saw the comma practise at all CSS properties in the file. Hence added
> it too.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/3337#discussion_r112802361>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKFeyt8h2xz3HD3hVito-JGenBKISOmqks5ryYAugaJpZM4M85bx>
> .
>

shaz13 commented:
@seanlip Oh right. My bad I misinterpret the question and thought I should add the selectors in directive css. Pushed the changes. Hope I got it right

shaz13 commented on assets/images/avatar/oppia_avatar_100px.svg:
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" preserveAspectRatio="xMidYMid meet" viewBox="0 0 100 100" width="100" height="100"><defs><path d="M98.32 49.97C98.4 76.67 76.84 98.42 50.22 98.5C23.6 98.57 1.92 76.96 1.84 50.26C1.76 23.55 23.31 1.81 49.94 1.73C76.56 1.65 98.24 23.27 98.32 49.97Z" id="f3kR7jUAgF"></path><path d="M81.58 50C81.58 69.41  ...
```
Done

seanlip commented:
OK, thanks @shaz13! @wxyxinyu, do you think this is good to go?

wxyxinyu commented:
Yep, looks good. To clarify, we'll replace files before the release with the actual otter avatars. This shouldn't affect anything if the file names are the same.
